### ISSUE TITLE
Syntax highlighting changes when `editable` is added

### ISSUE THREAD

**Issue by mgeisler** [rust-lang/mdBook#1796]
Hi there,

I noticed that the syntax highlighting changes for a Rust code block when it is made editable. The first code block is editable, the second is not:

![image](https://user-images.githubusercontent.com/89623/166233701-ea7a947d-a0c0-4795-b2d8-9fb745a1630f.png)

I guess part of this is because the Ace editor uses a different theme and/or a different syntax highlighting engine.

Would there be interest in changing the default theme for one or the other kind of code block so that the colors match between them?

From a quick skim of the generated HTML, it seems that the syntax highlighting is simpler in the Ace editor compared to the highlighting in non-editable blocks: `print!` is marked up as a builtin in the non-editable block, but it has no special class in the Ace editor highlighting.

**Comment by joaofreires** [comment]
They could deprecate the old highlight and fully adopt the Ace editor. Ace has a read-only mode that can be used and standardize highlights.

**Comment by mgeisler** [comment]
> Ace has a read-only mode that can be used and standardize highlights.

Yeah, it would be great to standardize this by using a read-only editor.

However, I'm not sure if the editor supports [hiding lines](https://rust-lang.github.io/mdBook/format/theme/syntax-highlighting.html#hiding-code-lines). Perhaps it could be worked around for read-only code blocks: since they're not editable, the hidden lines could even be removed from the generated output.

Doing so would probably imply deprecating/removing the eyeball icon which can be used to toggle the hidden lines. Personally, I would prefer that hidden lines stay hidden since they can often contain strange and unhelpful code which is only there to make the example compile. So I would like an option to not include the eyeball icon in the first place.

**Comment by joaofreires** [comment]
I think the [hiding lines](https://rust-lang.github.io/mdBook/format/theme/syntax-highlighting.html#hiding-code-lines) feature could be deprecated. It only works for rust code and after builds, generating incompatibilities between the [markdown version](https://github.com/rust-lang/mdBook/blob/master/guide/src/format/theme/syntax-highlighting.md#hiding-code-lines) and the [builded version](https://rust-lang.github.io/mdBook/format/mdbook.html?highlight=ANCHOR#hiding-code-lines).

**Comment by mgeisler** [comment]
> I think the [hiding lines](https://rust-lang.github.io/mdBook/format/theme/syntax-highlighting.html#hiding-code-lines) feature could be deprecated. It only works for rust code and after builds, generating incompatibilities between the [markdown version](https://github.com/rust-lang/mdBook/blob/master/guide/src/format/theme/syntax-highlighting.md#hiding-code-lines) and the [builded version](https://rust-lang.github.io/mdBook/format/mdbook.html?highlight=ANCHOR#hiding-code-lines).

Deprecating the hidden lines feature would also help solve #1767 which is about editable examples not compiling without a `fn main`.
